gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-server/src/main/java/org/apache/gobblin/service/FlowConfigResourceLocalHandler.java

```diff
@@ -199,4 +202,12 @@ public static FlowSpec createFlowSpecForConfig(FlowConfig flowConfig) {
       throw new FlowConfigLoggedException(HttpStatus.S_400_BAD_REQUEST, "bad URI " + flowConfig.getTemplateUris(), e);
     }
   }
+
+  /**
+   * @param flowConfig flowConfig
+   * @return returns true if it is a runOnce flow, i.e. runImmediately is true and cron schedule is empty
+   */
+  private static boolean isRunOnceFlow(FlowConfig flowConfig) {
```

Comment:
One one is defined as not having a schedule.

Reply:
Yes, run-once is defined as not having a schedule and as a run-immediately.
cronSchedule is compulsory field in Schedule, so it cannot be skipped, it can only be written as empty. Do you want to make it optional in Schedule.pdsc?

Reply:
No, the schedule is optional in the flowConfig. A schedule must have a valid cron expression.
`{
      "name" : "schedule",
      "type" : "Schedule",
      "optional" : true,
      "doc" : "Optional schedule for when to execution the flow. If a schedule is not specified then the flow is executed immediately."
    },`, so we should not make the cron optional in Schedule.pdsc.

Reply:
yea, but 'runImmediately' is a part of Schedule, so we need it.
Should we make the job runImmediately if Schedule is absent?